pytorch_lightning/trainer/trainer.py

```diff
@@ -581,7 +582,7 @@ def __init__(
 
         # training state
         self.model = None
-        self.testing = False
+        self.mode = TrainerMode.TRAINING
```

Comment:
We set `TrainerMode.TRAINING` mode only there, in the init of the trainer. It means, for example, that if the user will call `trainer.fit()` after `trainer.test()` this `fit()` will work in the testing mode. Possible solution: rename `fit()` to `_fit()` and add new `fit()` which will set `TrainerMode.TRAINING` and call `_fit()`.